tape_measure_tool.md

```diff
@@ -0,0 +1,90 @@
+
+I expect the user experience to be something like the following:
+ 1. The user adds the Tape Measure Tool plugin via the plugin menu.
+ 2. At this point, I suppose there could be a button one could click
+    that is something like "New measurement" at which point would enable
+    the tape measure tool.  There could also be a hotkey to start a new
+    measurement so the user can quickly take measurements without having
+    to constantly click the "New Measurement" button.
+ 3. The mouse could then change into a cross-shaped icon and display a
+    transparent point which gives a preview of the point to be measured,
+    this point will become opaque once the user clicks.
+ 4. We have two options at this point, neither of which would differ
+    in behavior, we could either:
+    * Have a click system where the user clicks point A and then clicks
+      point B.
+    * Have a drag system where the user clicks on a particular point, holding
+      down the mouse and dragging the cursor to the desired end point.
+ 5. When the user does either of the above two, points should appear at point
+    A and point B, and the user should see a marker line between the two
+    points, even while being in the process of choosing point B.
+ 6. The distance can be written in text along the line marker that appears
+    between the two points as well as written in a text box within the plugin
+    window.
+ 7. At any point during this process, the user can press escape or right click
+    to cancel the measurement.
+ 8. The user can hold control to snap to certain key points such as the
+    boundaries of an entity's bounding box, the corners of those bounding
+    boxes, or the grid lines as well.
+
+## UI Design
+
+I don't expect this one to be too complicated, here's a simple mock-up of
+something I made.  My GIMP skills aren't too great so this is just a simple
+prototype, color of the line marker and GUI layout are likely going to change
+and are open for suggestions.  I just chose yellow as it was the most
+contrasting color in the shapes scene.  Note that the "New Measurement"
+section is supposed to be a button.  After making this picture, I realized
+that an xyz axis placed at point A may also be helpful for the user.
+
+![Tape measure UI](images/tape_ui.png)
+
+## Nice to haves
+
+ * Display bounding boxes when hovering during the selection.
+ * Manipulating the end points once the tape measure is placed.
+ * Specifying the length of the tape so that it will lock to that length.
+ * Specifying an angle (relative to the x, y, or z axis, I suppose) that the
+   tape measure will lock to.
+ * Hold shift to snap and constrain the tape measure to the closest axis.
+
+## Questions of functionality
+
+ * Should we allow multiple measurements within the scene or only one?
+ * Should we attach the marker to the model/link if the user clicks one?
+   See [Louise's Gazebo Classic example here](https://www.youtube.com/watch?v=XjszkNSthok&feature=youtu.be)
```

Comment:
I like that and I think it can be useful to keep track of the distance between a robot and models moving around it, for example.